# Part 3A - EDA

**Student:** Matthew Malueg

## Imports and Loading

In [2]:
# import needed modules
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', 100)

In [3]:
# loading data
fpath = "tmdb_results_combined.csv.gz"
df = pd.read_csv(fpath)
df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /tJLV3BAlHOgscVOrA99Wnb2gAef.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0116916    0.0  /rFpHBidSlhjflmnLu7BZilyKeQR.jpg                   NaN   
4  tt0118589    0.0  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 10749, 'name': 'Romance'}, {'id': 14, ...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3         0.0  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      NaN   
4  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   

         id original_language    original_title  \
0       NaN               NaN               NaN   
1   11232.0                en    Kate & Leopold   
2  151007.0                en  The Silent Force   
3   73549.0                en     The Dark Mist   
4   10696.0                en           Glitter   

                                            overview  popularity  \
0                                                NaN         NaN   
1  When her scientist ex-boyfriend discovers a po...      13.382   
2  An elite Federal force whose objective is to e...       2.612   
3  With the future of the world hanging in the ba...       2.744   
4  A young woman is catapulted into pop stardom, ...       7.460   

                        poster_path  \
0                               NaN   
1  /mUvikzKJJSg9khrVdxK8kg3TMHA.jpg   
2  /u0njmfXD3dXxQfeykBHBuFQOYqD.jpg   
3  /z18E0lukzOtAALmfINSTCUU0Als.jpg   
4  /b8gKsXB6d60SC2HKUcAAzTacn0S.jpg   

                                production_companies  \
0                                                NaN   
1  [{'id': 85, 'logo_path': None, 'name': 'Konrad...   
2                                                 []   
3  [{'id': 71364, 'logo_path': None, 'name': 'Car...   
4  [{'id': 89627, 'logo_path': None, 'name': 'Gli...   

                                production_countries release_date     revenue  \
0                                                NaN          NaN         NaN   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   2001-12-25  76019048.0   
2                                                 []   2001-01-01         0.0   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1996-01-01         0.0   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   2001-09-21   5271666.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'French', 'iso_639_1': 'fr',...  Released   
2     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3    101.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4    104.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline             title  video  \
0                                                NaN               NaN    NaN   
1  If they lived in the same century, they'd be p...    Kate & Leopold    0.0   
2  They left him for dead... They should have fin...  The Silent Force    0.0   
3                                                NaN     The Dark Mist    0.0   
4   In music she found her dream, her love, herself.           Glitter    0.0   

   vote_average  vote_count certification  
0           NaN         NaN           NaN  
1         6.319      1238.0         PG-13  
2         5.000         3.0           NaN  
3         3.500         2.0            PG  
4         4.405       132.0         PG-13

## Objective questions

**Q1: How many movies had at least some valid financial information (values > 0 for budget OR revenue)?**

In [8]:
df=df.loc[(df['budget'] >0)|(df['revenue']>0)]
df.shape

(726, 26)

- There are 726 movies with some type of financial information available.

**Q2: How many movies are there in each certification category?**

In [12]:
df['certification'].value_counts()

R        246
PG-13    148
PG        42
NR        22
G         15
Name: certification, dtype: int64

- There are:
    - 246 R-rated movies
    - 148 PG-13 movies
    - 42 PG movies
    - 15 G movies

**Q3: What is the average revenue per certification category?**

In [51]:
# Filter out films with zero budget AND zero revenue
budget_filter = df.loc[(df['budget'] > 0)]
df_filtered = budget_filter.loc[(budget_filter['revenue'] > 0)]

In [55]:
# Group the filtered dataframe by certification
cert_df = df_filtered.groupby('certification')
cert_df

In [58]:
# Get the average values for each certification group
avg_rev = cert_df[['revenue', 'budget']].mean()
print(avg_rev)

                    revenue        budget
certification                            
G              1.401132e+08  4.781818e+07
NR             1.215925e+07  5.420000e+06
PG             1.901400e+08  5.064404e+07
PG-13          1.102883e+08  4.368342e+07
R              4.620896e+07  2.738208e+07


- The average revenue for each certification category is:
    - R: \$46,208,960
    - PG13: \$110,288,300
    - PG: \$190,140,000
    - G: \$140,113,200

**Q4: What is the average budget per certification category?**

- The average budget for each certification category is:
    - R: \$27,382,080
    - PG13: \$43,683,420
    - PG: \$50,644,040
    - G: \$47,818,180